docs/dev/cut-a-new-release.md

```diff
@@ -207,3 +207,81 @@ If you need to release a hotfix for a previously released version (that is: not
 
 On Android, Mozilla products consume the Glean SDK through its wrapper in [`android-components`](https://github.com/mozilla-mobile/firefox-android/tree/main/android-components/).
 The process of vendoring a new version of Glean in to Mozilla Central now handles the upgrade process for android-components and no manual updating is needed here.
+
+## Recovering from a failed automated release
+
+In rare circumstances the automated release on CI can fail to publish updated packages.
+This usually can be fixed with some manual steps.
+
+### CI re-run
+
+In some circumstances it's possible to re-run the CI tasks.
+This should be done if any of the tasks fail with an intermittent issues (e.g. broken network connectivity, package repository APIs unreachable, ...).
+
+Find the release task on <https://app.circleci.com/pipelines/github/mozilla/glean/> and select "Rerun workflow from failed" from the "Rerun" drop-down.
+Monitor the tasks for successful completion.
+
+### Manual release
+
+If any of the package releases fail for some other reason it might be possible to manually publish the package, using the same commands being run on CI.
+
+#### Manual release: Rust crates
+
+Make sure you're logged in to `crates.io`:
```

Comment:
Well...   if you have the proper access. Since I doubt anyone but us would need to do a manual release, this is probably fine as is, but it might not hurt to mention that you need to be an owner at the deployment site (pypi, crates.io, etc)